frontend/ui/presets.lua

```diff
@@ -0,0 +1,350 @@
+--[[
+    This module provides a unified interface for managing presets across different KOReader modules.
+    It handles creation, loading, updating, and deletion of presets, as well as menu generation.
+
+    Usage:
+        local Presets = require("ui/presets")
+
+        -- 1. Initialize preset configuration in your module's init() method:
+            self.preset_config = {
+                presets = G_reader_settings:readSetting("my_module_presets", {}),                -- or custom storage
+                cycle_index = G_reader_settings:readSetting("my_module_presets_cycle_index", 0), -- optional, only needed if cycling through presets
+                dispatcher_name = "load_my_module_preset", -- must match dispatcher.lua entry
+                save = function(config)
+                    G_reader_settings:saveSetting("my_module_presets", config.presets)
+                end,
+                saveCycleIndex = function(config)
+                    G_reader_settings:saveSetting("my_module_presets_cycle_index", config.cycle_index)
+                end,
+            }
+
+        -- 2. Implement required methods in your module:
+            function MyModule:buildPreset()
+                return {
+                    -- Return a table with the settings you want to save in the preset
+                    setting1 = self.setting1,
+                    setting2 = self.setting2,
+                    enabled_features = self.enabled_features,
+                }
+            end
+
+            function MyModule:loadPreset(preset)
+                -- Apply the preset settings to your module
+                self.setting1 = preset.setting1
+                self.setting2 = preset.setting2
+                self.enabled_features = preset.enabled_features
+                -- Update UI or perform other necessary changes
+                self:refresh()
+            end
+
+        -- 3. Create menu items for presets:
+            function MyModule:genPresetMenuItemTable(touchmenu_instance)
+                return Presets:genPresetMenuItemTable(
+                    self.preset_config,                              -- preset configuration object
+                    _("Create new preset from current settings"),    -- optional: custom text for create button
+                    function() return self:hasValidSettings() end,   -- optional: function to enable/disable create button
+                    function() return self:buildPreset() end,        -- function to build preset data
+                    function(preset) self:loadPreset(preset) end,    -- function to load preset data
+                    function()                                       -- callback when presets are updated
+                        touchmenu_instance.item_table = self:genPresetMenuItemTable(touchmenu_instance)
+                        touchmenu_instance:updateItems()
+                    end
+                )
+            end
+
+        -- 4. Create a new preset programmatically:
```

Comment:
Everything is programatically :)
Reading this, I though this would allow creating a setting with programatilcally created settings (which seems uneeded).
Shouldn't it read: `4. Create a new preset from current settings` ? Oh, that's what the function title names says :)